R/RichDEM/build_tarballs.jl

```diff
@@ -17,24 +22,46 @@ sources = [
 script = raw"""
 cd $WORKSPACE/srcdir/richdem-*
 mkdir build && cd build
-cmake -DJulia_PREFIX=$prefix -DCMAKE_FIND_ROOT_PATH=$prefix -DJlCxx_DIR=$prefix/lib/cmake/JlCxx -DCMAKE_INSTALL_PREFIX=$prefix -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} -DCMAKE_BUILD_TYPE=Release -DUSE_GDAL=ON ../. 
-cmake --build . --config Release --target install 
+
+cmake \
+    -DJulia_PREFIX=$prefix \
+    -DCMAKE_FIND_ROOT_PATH=$prefix \
+    -DJlCxx_DIR=$prefix/lib/cmake/JlCxx \
+    -DCMAKE_INSTALL_PREFIX=$prefix \
+    -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} \
+    -DCMAKE_BUILD_TYPE=Release \
+    -DUSE_GDAL=ON ../. 
+
+# NetCDF is the most restrictive dependency as far as platform availability, so we'll use it where applicable but disable it otherwise
+if ! find ${libdir} -name "libnetcdf*.${dlext}" -exec false '{}' +; then
+    CMAKE_FLAGS+=(-DWITH_NETCDF=ON)
+else
+    echo "Disabling NetCDF support"
+    CMAKE_FLAGS+=(-DWITH_NETCDF=OFF)
+fi
+# HDF5 is also a restrictive dependency as far as platform availability, so we'll use it where applicable but disable it otherwise
+if ! find ${libdir} -name "libhdf5*.${dlext}" -exec false '{}' +; then
+    CMAKE_FLAGS+=(-DWITH_HDF5=ON)
+else
+    echo "Disabling HDF5 support"
+    CMAKE_FLAGS+=(-DWITH_HDF5=OFF)
+fi
+if [[ "${target}" == x86_64-linux-musl* ]]; then
+    export LDFLAGS="$LDFLAGS -lcurl"  # same fix as used for PROJ
+    rm /usr/lib/libexpat.so.1  # ugly, but can't figure out CMake behaviour here
+fi
+
+VERBOSE=ON cmake --build . --config Release --target install -- -j${nproc} 
 """
 
 # These are the platforms we will build for by default, unless further
 # platforms are passed in on the command line
-platforms = [
-    Platform("i686", "linux"; libc = "glibc"),
-    Platform("x86_64", "linux"; libc = "glibc"),
-    Platform("aarch64", "linux"; libc = "glibc"),
-    Platform("armv6l", "linux"; call_abi = "eabihf", libc = "glibc"),
-    Platform("armv7l", "linux"; call_abi = "eabihf", libc = "glibc"),
-    Platform("powerpc64le", "linux"; libc = "glibc"),
-    Platform("aarch64", "linux"; libc = "musl"),
-    Platform("armv6l", "linux"; call_abi = "eabihf", libc = "musl"),
-    Platform("armv7l", "linux"; call_abi = "eabihf", libc = "musl"),
-]
-
+julia_versions = [v"1.6.3", v"1.8"]
+include("../../L/libjulia/common.jl")
+platforms = vcat(libjulia_platforms.(julia_versions)...)
+platformfilter(p) = (p.tags["arch"] != "armv6l" && p.tags["os"] ∉ ["macos", "freebsd"])
```

Comment:
Avoid accessing tags, we have functions
```suggestion
platformfilter(p) = (arch(p) != "armv6l" && Sys.isbsd(p))
```